src/lib/modernExtend.ts

```diff
@@ -358,17 +381,26 @@ export function numeric(args: NumericArgs): ModernExtend {
         },
     }];
 
-    return {exposes: [expose], fromZigbee, toZigbee, isModernExtend: true};
+    const configure: Configure = async (device, coordinatorEndpoint, logger) => {
+        // call setupAttributes if we have a configureReporting payload or configureSkipRead is not set, but then use readOnly=true
+        const ep = device.getEndpoint(isNumber(configureEndpointId) ? configureEndpointId : 1);
```

Comment:
The `configureEndpointId` option is not needed, the `setupAttributes` already configures it on the correct endpoint.

Reply:
Sadly that does not work for Xiaomi/Aqara devices ... see the 2nd point of this comment: https://github.com/Koenkk/zigbee-herdsman-converters/issues/6721#issuecomment-1858909096

We should probably not default to 1 though, and let it figure it out when unset.

Reply:
Just double check, without specifying ep1 it indeed fails but only for the aqara device I have. It works fine for a sonoff SNZB02 I had also temporary added a numeric modernExtend too.

```
error 2023-12-17 18:04:59: Failed to configure 'airquality/livingroom', attempt 1 (Error: Device 0x54ef441000684954 has no input cluster aqaraOpple
    at getEndpointsWithInputCluster (/opt/zigbee2mqtt/node_modules/zigbee-herdsman-converters/src/lib/modernExtend.ts:23:15)
    at setupAttributes (/opt/zigbee2mqtt/node_modules/zigbee-herdsman-converters/src/lib/modernExtend.ts:34:55)
    at configure (/opt/zigbee2mqtt/node_modules/zigbee-herdsman-converters/src/lib/modernExtend.ts:323:17)
    at Object.configure (/opt/zigbee2mqtt/node_modules/zigbee-herdsman-converters/src/index.ts:138:31)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at Configure.configure (/opt/zigbee2mqtt/lib/extension/configure.ts:117:13)
    at Configure.onMQTTMessage (/opt/zigbee2mqtt/lib/extension/configure.ts:55:21))
```

I will leave configureEndpointId in place and set it to 1 in src/lib/xiaomi.ts for those 3 modernExtends but have it default to grabbing all the endpoints.